chat/src/main/java/net/md_5/bungee/api/chat/TextComponent.java

```diff
@@ -111,15 +111,15 @@ public static BaseComponent[] fromLegacyText(String message, ChatColor defaultCo
                 } else if ( format == ChatColor.MAGIC )
                 {
                     component.setObfuscated( true );
-                } else if ( format == ChatColor.RESET )
-                {
-                    format = defaultColor;
-                    component = new TextComponent();
-                    component.setColor( format );
                 } else
                 {
+                    if ( format == ChatColor.RESET )
+                    {
+                        format = defaultColor;
+                    }
                     component = new TextComponent();
                     component.setColor( format );
+                    component.setReset( true );
```

Comment:
should be 4 lines up right?

Reply:
If previous component is bold or italic, won't this then remove those formattings from the new color component?